src/mix_protocol.nim

```diff
@@ -258,35 +248,75 @@ proc anonymizeLocalProtocolSend*(
     error "Failed to initialize PeerId", err = error
     return
 
+  trace "# Sending to: ", multiaddr = multiAddrs[0]
+
   var nextHopConn: Connection
   try:
     nextHopConn =
       await mixProto.switch.dial(firstMixPeerId, @[firstMixAddr], @[MixProtocolID])
     await nextHopConn.writeLp(sphinxPacket)
-    await nextHopConn.close()
   except CatchableError as e:
     error "Failed to send message to next hop: ", err = e.msg
+  finally:
+    if nextHopConn != nil:
+      try:
+        await nextHopConn.close()
+      except CatchableError as e:
+        error "Failed to close outgoing stream: ", err = e.msg
+
+proc createMixProtocol*(
+    mixNodeInfo: MixNodeInfo,
+    pubNodeInfo: Table[PeerId, MixPubInfo],
+    switch: Switch,
+    tagManager: TagManager,
+    handler: ProtocolHandler,
+): Result[MixProtocol, string] =
+  let mixProto = new MixProtocol
+  mixProto.mixNodeInfo = mixNodeInfo
+  mixProto.pubNodeInfo = pubNodeInfo
+  mixProto.switch = switch
+  mixProto.tagManager = tagManager
+  mixProto.pHandler = handler
+  mixProto.init()
+
+  return ok(mixProto)
```

Comment:
Need to home in on the code that this helper fn is replacing, and apply this refactor to the waku side of things.